Add support for markdownlint

Add problem matcher for [markdownlint](https://github.com/igorshubovych/markdownlint-cli).

Example output from markdownlint:

```
2022-08-29 02:13:21 [INFO]   
2022-08-29 02:13:21 [INFO]   ----------------------------------------------
2022-08-29 02:13:21 [INFO]   ----------------------------------------------
2022-08-29 02:13:21 [INFO]   Linting [MARKDOWN] files...
2022-08-29 02:13:21 [INFO]   ----------------------------------------------
2022-08-29 02:13:21 [INFO]   ----------------------------------------------
2022-08-29 02:13:21 [INFO]   ---------------------------
2022-08-29 02:13:21 [INFO]   File:[/github/workspace/README.md]
2022-08-29 02:13:21 [ERROR]   Found errors in [markdownlint] linter!
2022-08-29 02:13:21 [ERROR]   Error code: 1. Command output:
------
/github/workspace/README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "Hi"]
```

--- a/README.md
+++ b/README.md
@@ -22,6 +22,7 @@ for common linters (see [list](#available-problem-matchers) below).
 * flake8_severity
 * isort
 * isort_severity
+* markdownlint
 * mypy
 * mypy_severity
 * pycodestyle

--- a/matchers/markdownlint.json
+++ b/matchers/markdownlint.json
@@ -1,0 +1,18 @@
+{
+    "problemMatcher": [
+        {
+            "owner": "markdownlint",
+            "severity": "error",
+            "pattern": [
+                {
+                    "regexp": "^(.+?):(\\d+)(?::(\\d+))? (MD\\d+)(?:/\\S+)? (.+)$",
+                    "file": 1,
+                    "line": 2,
+                    "column": 3,
+                    "code": 4,
+                    "message": 5
+                }
+            ]
+        }
+    ]
+}

